File: frontEnd/src/app/component/models/LaporanHeader.ts
import { Guid } from "guid-typescript";
import { User } from "src/app/models/UserModel";
import { LaporanItem } from "./LaporanItem";

export class LaporanHeader {
    laporanHeaderId: Guid = Guid.createEmpty();
    judulLaporan: string = "";
    keterangan: string = "";
    statusLaporan: string = "";
    userId: string = "";
    tglBuat: Date = null;
    tglSelesai: Date = null;
    user: User = new User;
    laporanItems: LaporanItem[] = [];
    status: string = "";
}

File: backEnd/Models/LaporanHeader.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace backEnd.Models
{
    public class LaporanHeader
    {
        public Guid LaporanHeaderId {get; set;}
        public string JudulLaporan {get; set;}
        public string Keterangan {get; set;}
        public string StatusLaporan {get; set;}
        public Guid UserId {get; set;}
        public DateTime TglBuat {get; set;}
        public DateTime? TglSelesai {get; set;}

        // NOTE INI BAGIAN RELASI DATABASE YANG NANTI DI PAKE INCLUDE
        public User User {get; set;}
        public List<LaporanItem> LaporanItems { get; set;}

        [NotMapped] public string Status { get; set; }
    }
}